drivers/hmis/app/graphql/types/hmis_schema/inventory_input.rb

```diff
@@ -0,0 +1,30 @@
+module Types
+  class HmisSchema::InventoryInput < BaseInputObject
+    description 'HMIS Inventory input'
+
+    argument :project_id, ID, required: false
```

Comment:
There probably isn't a way to do this, but I sure wish these arguments could be built from the fields on the inventory.rb file.  It seems like there should be a mechanism to setup defaults to match what's in inventory and then add/remove any that aren't needed.  As it is, it's going to be a bit of a chore each time the spec changes to go through 2+ places and make sure the sets align.

Reply:
I had a better idea!
In `drivers/hmis/app/graphql/types/hmis_schema/inventory.rb`
```
FIELDS = {
 household_type: { type: HmisSchema::Enums::HouseholdType, null: false, required: false, input_argument: true },
 availability:  { type: HmisSchema::Enums::Availability, null: true, required: false, input_argument: true }, 
...
}
FIELDS.each do |column, options|
  field column, options[:type], null: options[:null]
end
```

in `drivers/hmis/app/graphql/types/hmis_schema/inventory_input.rb`
```
FIELD_SOURCE=Types::HmisSchema::Inventory
FIELD_SOURCE::FIELDS.each do |column, options|
  next unless options[:input_argument]

  argument column, options[:type], required: options[:required]
end
```

That way, we can collect up a bunch of the logic in the FIELDS constant and use it throughout.  
Also, the looping over Fields can for both input and the base class can be moved to a concern or base class.

Reply:
Thanks for coming up with this! It's definitely painful to need to update fields in multiple places. I will give this a go.